logprep/connector/s3/output.py

```diff
@@ -195,52 +198,57 @@ def _write_to_s3_resource(self, document: dict, prefix: str):
         ----------
         document : dict
            Document to store.
-
-        Returns
-        -------
-        Returns True to inform the pipeline to call the batch_finished_callback method in the
-        configured input
         """
         prefix = self._add_dates(prefix)
         prefix = f"{self._base_prefix}{prefix}"
         self._message_backlog[prefix].append(document)
 
-        backlog_count = self._current_backlog_count + 1
-        if backlog_count == self._config.message_backlog_size:
-            if self._writing_thread is None or not self._writing_thread.is_alive():
-                message_backlog = deepcopy(self._message_backlog)
-                self._writing_thread = threading.Thread(
-                    target=self._write_document_batches, args=(message_backlog,)
-                )
-                self._writing_thread.start()
-                return True
-        self._current_backlog_count = backlog_count
-        return False
-
-    def _write_document_batches(self, message_backlog):
-        self._logger.info(f"Writing {self._current_backlog_count + 1} documents to s3")
-        for prefix_mb, document_batch in message_backlog.items():
+        if self._get_backlog_size(self._message_backlog) >= self._config.message_backlog_size:
+            self._write_backlog()
+
+    @staticmethod
+    def _get_backlog_size(message_backlog: dict):
+        return sum(len(values) for values in message_backlog.values())
```

Comment:
```suggestion
    @property
    def _backlog_size():
        return sum(map(len, self._message_backlog.values()))
```
how about somethin like this, to avoid looping? And to make access to this easier?
not testet, but should do the trick...